File: frontend/src/lib/api/notifications.ts
import { apiGet, apiPatch } from './client';
import type { Page } from './chatbot';

export type NotificationCategory = 'alert' | 'incident' | 'vulnerability' | 'system';
export type NotificationSeverity = 'info' | 'warning' | 'critical';

export interface NotificationRecord {
  id: string;
  title: string;
  body: string;
  category: NotificationCategory;
  severity: NotificationSeverity;
  is_read: boolean;
  source_ref: string;
  created_at: string;
  updated_at: string;
}

export interface NotificationPage extends Page<NotificationRecord> {
  unread_count: number;
}

export function listNotifications(unreadOnly = false): Promise<NotificationPage> {
  const params = new URLSearchParams();
  params.set('page', '1');
  params.set('page_size', '100');
  if (unreadOnly) params.set('unread_only', 'true');
  return apiGet<NotificationPage>(`/api/notifications?${params.toString()}`);
}

export function setNotificationRead(id: string, isRead: boolean): Promise<NotificationRecord> {
  return apiPatch<NotificationRecord>(`/api/notifications/${id}/read`, { is_read: isRead });
}

File: backend/schemas/notifications.py
"""Schemas for the notification center."""

from typing import Literal
from uuid import UUID

from pydantic import BaseModel, Field

from backend.schemas.common import Page, UtcDatetime

NotificationCategory = Literal["alert", "incident", "vulnerability", "system"]
NotificationSeverity = Literal["info", "warning", "critical"]


class NotificationCreate(BaseModel):
    title: str = Field(min_length=1, max_length=240)
    body: str = Field(default="", max_length=4000)
    category: NotificationCategory
    severity: NotificationSeverity = "info"
    source_ref: str = Field(default="", max_length=200)


class NotificationItem(BaseModel):
    id: UUID
    title: str
    body: str
    category: NotificationCategory
    severity: NotificationSeverity
    is_read: bool
    source_ref: str
    created_at: UtcDatetime
    updated_at: UtcDatetime


class NotificationPage(Page[NotificationItem]):
    unread_count: int


class NotificationReadUpdate(BaseModel):
    is_read: bool
